http_server/queries/staff/ban_user.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+function throttle_bans($pdo, $mod_user_id) {
+	$time = (int) (time() - 3600);
+  
+	$stmt = $pdo->prepare('SELECT COUNT(*) as recent_ban_count FROM bans WHERE mod_user_id = ? AND time > ?');
+	$stmt->bindValue(1, $mod_user_id, PDO::PARAM_INT);
+	$stmt->bindValue(2, $time, PDO::PARAM_INT);
+	$stmt->execute();
+	$row = $stmt->fetchAll(PDO::FETCH_OBJ);
+	return $row;
+}
+
+function ban_user($pdo, $banned_ip, $banned_user_id, $mod_user_id, $expire_time, $reason, $record, $banned_name, $mod_name, $ip_ban, $account_ban) {
+	$time = (int) time();
+	
+	// the SQL we want
+	$stmt = $pdo->prepare('INSERT INTO bans (banned_ip, banned_user_id, mod_user_id, time, expire_time, reason, record, banned_name, mod_name, ip_ban, account_ban) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)');
+
+	// bind the parameters
+	$stmt->bindValue(1, $banned_ip, PDO::PARAM_STR);
+	$stmt->bindValue(2, $banned_user_id, PDO::PARAM_INT);
+	$stmt->bindValue(3, $mod_user_id, PDO::PARAM_INT);
+	$stmt->bindValue(4, $time, PDO::PARAM_INT);
+	$stmt->bindValue(5, $expire_time, PDO::PARAM_INT);
+	$stmt->bindValue(6, $reason, PDO::PARAM_STR);
+	$stmt->bindValue(7, $record, PDO::PARAM_STR);
+	$stmt->bindValue(8, $banned_name, PDO::PARAM_STR);
+	$stmt->bindValue(9, $mod_name, PDO::PARAM_STR);
+	$stmt->bindValue(10, $ip_ban, PDO::PARAM_INT);
+	$stmt->bindValue(11, $account_ban, PDO::PARAM_INT);
+	
```

Comment:
What if we use named parameters if there are more than a few question mark placeholders? It seems like it that would help prevent leaving something out or getting the order wrong.

```
VALUES (:banned_id, :banned_user_id, ...);
$stmt->bindValue(':banned_id', $banned_ip, PDO::PARAM_STR);
$stmt->bindValue(':banned_user_id', $banned_user_id, PDO::PARAM_INT);
```

Reply:
Yeah I can do that, let me edit this real quick.

Reply:
done!